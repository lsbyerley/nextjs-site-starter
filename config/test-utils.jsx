// test-utils.js
import { render } from '@testing-library/react';

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers = ({ children }) => {
  return children;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export {
  default as userEvent,
  specialChars,
} from '@testing-library/user-event';
// export * from 'next-page-tester';

// override render method
export { customRender as render };
