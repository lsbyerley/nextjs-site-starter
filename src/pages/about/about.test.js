import React from 'react';
import { render } from 'test-utils';
import About from './Index';

jest.mock('next/image', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation((props) => {
    return <div role='image'>{JSON.stringify(props)}</div>;
  }),
}));

const renderAbout = (props) => {
  const utils = render(<About {...props} />);
  const about = utils.container.firstChild;
  return { ...utils, about };
};

describe('About', () => {
  it('renders and matches snapshot', () => {
    const { about } = renderAbout();
    expect(about).toBeTruthy();
    expect(about).toMatchSnapshot();
  });
});
