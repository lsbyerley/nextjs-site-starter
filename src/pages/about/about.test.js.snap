// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders and matches snapshot 1`] = `
<div
  class="space-y-14 lg:space-y-24"
>
  <main
    class="max-w-4xl mx-auto mt-16"
  >
    <div
      class="container px-4 mx-auto"
    >
      <div
        class="flex flex-col-reverse text-center lg:space-x-5 lg:flex lg:flex-row item-center lg:-mx-4 lg:text-left"
      >
        <div
          class="lg:px-4 lg:mt-12 "
        >
          <h1
            class="text-2xl font-bold text-gray-900 lg:text-5xl dark:text-white"
          >
            The About page,
          </h1>
          <div
            class="mt-6 text-gray-800 dark:text-white"
          >
            <p
              class="mb-4"
            >
              This page is only here to show creating another page and testing it. Feel free to remove it if it serves no purpose.
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
